src/Framework/FeatureAvailabilityChecker.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// The status of a feature.
+    /// </summary>
+    public enum FeatureStatus
+    {
+        /// <summary>
+        /// The feature is not found.
+        /// </summary>
+        NotFound,
+
+        /// <summary>
+        /// The feature is available.
+        /// </summary>
+        Available,
+
+        /// <summary>
+        /// The feature is not available.
+        /// </summary>
+        NotAvailable,
+    }
+
+    /// <summary>
+    /// This class is used to check if a feature is available or not.
+    /// </summary>
+    public static class FeatureAvailabilityChecker
```

Comment:
I expect this class might grow in scope by adding like list of features and such.
Please consider to rename it along with comment. I'd suggest something generic such as 'Features'